File: Clean_USACO/training/python/moocast.py
"""
ID: agentmz1
LANG: PYTHON3
TASK: test
NOTES:
"""
#-- IO
__taskname__ = 'test'
fin = open(__taskname__ + '.in', 'r')
fout = open(__taskname__ + '.out', 'w')

#-- Types
class Point:
    def __init__(self, x, y, r):
        self.x = x
        self.y = y
        self.range = r
    
    def __str__(self):
        return "(%d, %d) --> %d" % (x, y, self.range)

#-- Functions
def dist(a, b):
    return (a.x-b.x)**2+(a.y-b.y)**2

#-- Var
n = eval(fin.readline())
data = []
for _ in range(4):
    x, y, r = map(int, fin.readline().split())
    data.append(Point(x, y, r))
    print(data[_])
for i in range(len(data)):
    for j in range(len(data)):
        if i==j:
            continue
        print("[%s] -> [%s], Dist %.1f, Can %sreach" % (data[i], data[j], dist(data[i], data[j]), "" if dist(data[i], data[j]) <= data[i].range+data[j].range else "not "))
#-- Run
#-- Exit

File: Clean_USACO/training/java/moocast.java
// package training;

/**
 * NAME: agentmz1
 * LANG: JAVA
 * PROG: BLANK
 * NOTES:
 */

import java.io.*;
import java.util.*;

public class moocast {
	private static final String programName = "test";

    //-- var dec

	public static void main(String[] args) throws IOException {
		BufferedReader stdin = new BufferedReader(new FileReader(programName + ".in"));
        PrintWriter stdout = new PrintWriter(new BufferedWriter(new FileWriter(programName + ".out")), true);//System.err, true);
        StringTokenizer read = new StringTokenizer(stdin.readLine());
        //-- init
        //-- run
        //-- exit
        stdin.close();
        stdout.close();
	}
    
    public double dist(Point a, Point b) {
        return (a.x-b.x)*(a.x-b.x)+(a.y-b.y)*(a.y-b.y);
    }

    static int get(StringTokenizer k, int l) {
        return Integer.parseInt(k.nextToken());
    }
    static String get(StringTokenizer k) {
        return k.nextToken();
    }
    static long get(StringTokenizer k, int l, int j) {
        return Long.parseLong(k.nextToken());
    }
}

class Point {
    int x, y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }
}